ground_segmentation:
  ros__parameters:
    # pre processing
    target_frame: "base_link"
    #target_frame: "velodyne_top"
    #target_frame: "husky"
    maxX:  20.0
    minX: -20.0
    maxY:  10.0
    minY: -10.0
    maxZ:  5.0
    minZ: -5.0
    downsample: true
    downsample_distance: 0.1

    # ground detection
    grid_type: 0
    radialCellSize: 2.0
    angularCellSize: 0.349066
    cellSizeX: 2.0
    cellSizeY: 2.0
    cellSizeZ: 2.0
    gridmaxX: 20.0
    gridmaxY: 10.0 
    gridmaxZ: 5.0
    startCellDistanceThreshold: 8.0
    slopeThresholdDegrees: 30.0
    groundInlierThreshold: 0.1
    returnGroundPoints: true
    neighborsIndexThreshold: 1
    minPoints: 5
    ransac_iterations: 50